rfcs/experimental/blocked_nd_range_ctad/README.md

```diff
@@ -0,0 +1,219 @@
+1. Be interpreted as one-dimensional range _[10, 20)_ (with grainsize 1 or 5). In this case it should be deduced as `blocked_nd_range<int, 1>` and 
+   constructed using the constructor `[1]`.
+2. Be interpreted as two (or three) dimensional range _[0, 10)_, _[0, 20)_ (and _[0, 5)_). In this case it should be deduced as `blocked_nd_range<int, 3>` 
+   and constructed using the constructor `[2]`. 
+
+Since it is unclear which resolution should be chosen, current proposal is not to support such use-case in CTAD and require the user to either explicitly
+specialize the template arguments, or to use array or `blocked_range` type itself to initialize the object.
+
+### Passing single C-array object of size 2 or 3
+
+Another interesting issue that should be resolved, is passing the single C-array object of size 2 or 3 to the constructor:
+
+```cpp
+int array[2] = {100, 200};
+tbb::blocked_nd_range range(array);
+```
+
+Since the `blocked_range` is not constructible from C-array and the braced-init-list is not used, the user expects the range to be deduced as
+`blocked_nd_range<int, 2>` and the constructor `[2]` to be used.
+
+If we add one more explicit deduction guide to support the code above, the single braced-init-list of size 2 or 3 would also match on this guide.
+
+There are the following options how this issue can be resolved:
+* Add a new deduction guide to support the code above. The downside of this approach is that it makes the ambiguity, discussed in the
+  [previous section](#ambiguity-while-passing-the-single-braced-init-list-of-size-2-or-3) to be resolved also and always result in 2 or 3-dimensional
+  range. If the user provided the single braced-init-list to have one-dimensional range, he would face the unexpected behavior without any diagnostics.
+* Document the code above as limitation and do not support it. The downside is  that the code above is considered valid, but    
+  cannot be supported because of the implementation of CTAD and current set of constructors.
+* Support the use-case above but do not support CTAD for braced-init-lists at all. The major downside is that the user would 
+  need to always specify the exact type `tbb::blocked_range` while using the braced-init-list construction.
+
+### Using the constructor `[1]` with "mixed" arguments
+
+There is a limitation of the deduction guides proposed if the constructor `[1]` is used with both arguments of exact `tbb::blocked_range` type
+and the braced-init-lists:
+
+```cpp
+tbb::blocked_range<int> dim_range(0, 100);
+tbb::blocked_nd_range nd_range(dim_range, {0, 200}, {0, 300}, dim_range);
+```
+
+These arguments would not match nether on the `[g1]` not `[g2]` and it is unclear how to define the deduction guide that covers this case.
+Current proposal is to keep this scenario a limitation for using the CTAD and always require using the consistent set of parameters - or 
+the set of braced-init-lists or the set of `tbb::blocked_range` objects.
```

Comment:
Can you add exit criteria? Likely the exit conditions are that we get feedback on user experience and it confirms choices made about open questions. This could be a place to collect those choices. And another exit criteria is that we get specification updates accepted (likely backed by user experience).

Reply:
Done